tests/config/test_config.py

```diff
@@ -11,22 +12,36 @@
 import pytest
 
 from poetry.config.config import Config
+from poetry.config.config import boolean_normalizer
+from poetry.config.config import int_normalizer
 
 
 if TYPE_CHECKING:
     from pathlib import Path
 
 
-def get_boolean_options(config: Optional[Dict[str, Any]] = None) -> str:
+def flatten(config: Optional[Dict[str, Any]], parent_key: str = "", sep="."):
```

Comment:
The type signature here is wrong (config cannot be `None`), and I'm not really a fan of this non-generator implementation. I think we should just add a dev dependency on [FlatDict](https://flatdict.readthedocs.io/en/stable/index.html) and use it in place of this custom function.

Reply:
I've added FlatDict, however, I'm also not a big fan of including a package just for this single test case.

Reply:
Yeah, I could go either way on that as well. However, we already do so with Zipp, Deepdiff, etc, and it's a common pattern in general and likely useful to have around to prevent additional bespoke implementations.